### PR TITLE
Check go modules consistency on CI and pin dev deps

### PR DESCRIPTION
**Pull request by danielfurman**
Issue: KNTK-316

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -77,4 +77,4 @@ Release process for the SDK is based on Git repository tags that follow [semanti
 
 To release the SDK:
 1. Make sure that all code that you want to release is in _master_ branch.
-1. Navigate to [repository releases page](https://github.com/kentik/community_sdk_golang/releases), click _Draft a new release_ button and put tag version (in _v[0-9].[0-9].[0-9]_ format), name and description.
+1. Navigate to [repository releases page](https://github.com/kentik/community_sdk_golang/releases), click _Draft a new release_ button and put tag version (in _v\[0-9].\[0-9].\[0-9]_ format), name and description.
```
Comment (mateuszmidor) [inline README.md:80]: Why the back-slashes?

File: tools/utility_functions.sh [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+function colored_echo() {
+    local color="$1"
+    local msg="$2"
+
+    if [[ ${BASH_VERSINFO[0]} -lt 4 ]]; then
+        # Bash 3.2 distributed with macOS does not support control sequences in echo -e
+        color_code=""
+    else
+        # shellcheck disable=SC2034
+        {
```
Comment (mateuszmidor) [inline tools/utility_functions.sh:23]: Why the curly brackets here?

File: README.md (hunk 1 above)
Comment (danielfurman) [reply README.md:80]: To escape special character `[` and get rid of warnings in Goland's markdown parser. I could have probably escaped also closing brackets, but I didn't need to.

See: https://daringfireball.net/projects/markdown/syntax -> Backslash Escapes

File: tools/utility_functions.sh (hunk 2 above)
Comment (danielfurman) [reply tools/utility_functions.sh:23]: So that `shellcheck disable` applies to all variables within those brackets.

**Review by mateuszmidor — APPROVED** [review]

**Review by mmac-m3a — APPROVED** [review]

**Pull request merged**